narwhals/_pandas_like/group_by.py

```diff
@@ -237,43 +258,58 @@ def agg_pandas(  # noqa: PLR0915
             new_names = [new_names[i] for i in index_map]
             result_simple_aggs.columns = new_names
 
+            result_aggs.append(result_simple_aggs)
+
         if nunique_aggs:
             result_nunique_aggs = grouped[list(nunique_aggs.values())].nunique(
                 dropna=False
             )
             result_nunique_aggs.columns = list(nunique_aggs.keys())
-        if simple_aggs and nunique_aggs:
-            if (
-                set(result_simple_aggs.columns)
-                .difference(keys)
-                .intersection(result_nunique_aggs.columns)
-            ):
+
+            result_aggs.append(result_nunique_aggs)
+
+        if std_aggs:
+            result_aggs.extend(
+                [
+                    grouped[std_root_names]
+                    .std(ddof=ddof)
+                    .set_axis(std_output_names, axis="columns", copy=False)
```

Comment:
@MarcoGorelli how bad is it to use `set_axis` to rename the columns? What's the alternative? Our `rename` does a mapping, yet here we could do:

```py
grouped[["b", "b"]].std(ddof=2).set_axis(["c", "d"], axis="columns")
```
For once we are exploiting some pandas weirdness 😁

Reply:
Should be fine let's just use copy=false